cli/Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift

```diff
@@ -66,9 +66,9 @@ final class BuildPhaseGenerator: BuildPhaseGenerating {
         fileElements: ProjectFileElements,
         pbxproj: PBXProj
     ) async throws {
-        if target.shouldIncludeHeadersBuildPhase, let headers = target.headers {
+        if shouldAddHeadersBuildPhase(target) {
```

Comment:
The headers build phase needs to be present in the target, otherwise the headers in the buildable folder are not picked